stripe/src/test/java/com/stripe/android/paymentsheet/PaymentSheetActivityTest.kt

```diff
@@ -120,24 +121,62 @@ internal class PaymentSheetActivityTest {
     }
 
     @Test
-    fun `updates buy button state`() {
+    fun `updates buy button state on payments list`() {
         val scenario = activityScenario()
         scenario.launch(intent).use {
             it.onActivity { activity ->
-                assertThat(activity.viewBinding.buyButton.isEnabled)
-                    .isTrue()
+                assertThat(activity.viewBinding.buyButton.isVisible).isTrue()
+                assertThat(activity.viewBinding.buyButton.isEnabled).isTrue()
 
                 viewModel.updateSelection(PaymentSelection.GooglePay)
-                assertThat(activity.viewBinding.buyButton.isEnabled).isTrue()
+                assertThat(activity.viewBinding.buyButton.isVisible).isFalse()
+                assertThat(activity.viewBinding.googlePayButton.isVisible).isTrue()
+                assertThat(activity.viewBinding.googlePayButton.isEnabled).isTrue()
 
                 viewModel.updateSelection(null)
+                assertThat(activity.viewBinding.buyButton.isVisible).isTrue()
                 assertThat(activity.viewBinding.buyButton.isEnabled).isFalse()
 
                 scenario.moveToState(Lifecycle.State.DESTROYED)
             }
         }
     }
 
+    @Test
+    fun `updates buy button state on add payment`() {
+        val scenario = activityScenario()
+        scenario.launch(intent).use {
+            it.onActivity { activity ->
+                assertThat(activity.viewBinding.buyButton.isVisible).isTrue()
+                assertThat(activity.viewBinding.buyButton.isEnabled).isTrue()
+
+                viewModel.transitionTo(
+                    PaymentSheetViewModel.TransitionTarget.AddPaymentMethodFull(
+                        FragmentConfigFixtures.DEFAULT
+                    )
+                )
+
+                assertThat(activity.viewBinding.buyButton.isVisible).isTrue()
+                assertThat(activity.viewBinding.buyButton.isEnabled).isFalse()
+                assertThat(activity.viewBinding.googlePayButton.isVisible).isFalse()
+
+                viewModel.updateSelection(PaymentSelection.GooglePay)
+                assertThat(activity.viewBinding.buyButton.isVisible).isTrue()
+                assertThat(activity.viewBinding.buyButton.isEnabled).isFalse()
+                assertThat(activity.viewBinding.googlePayButton.isVisible).isFalse()
+                viewModel.onGooglePayResult(StripeGooglePayContract.Result.Canceled)
+
+                viewModel.updateSelection(
+                    PaymentSelection.Saved(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
```

Comment:
If you are testing each of the PaymentSelection types, I think PaymentSelection.New.Card is the other one.

Reply:
Done.